cpp/ql/src/Security/CWE/CWE-676/PotentiallyDangerousFunction.ql

```diff
@@ -2,7 +2,7 @@
  * @name Use of potentially dangerous function
  * @description Certain standard library functions are dangerous to call.
```

Comment:
I think this name and description has always been a bit exaggerated. Now that we're toning down the severity, let's also tone down the other metadata. Perhaps we can focus this query on non-reentrant functions. Then the name could be "Use of non-reentrant function" or "Use of non-threadsafe function", and the description could be adjusted similarly. Over time we can add [other such functions](https://stackoverflow.com/a/6484929) to the query.

I think it makes sense to focus the query on re-entrancy since all re-entrancy issues will share the same alert suppression patterns: if someone wants to suppress results about non-reentrant _time_ functions in their `lgtm.yml`, they probably also want to suppress results about non-reentrant _DNS lookup_ functions. If we add logic to the query to only produce alerts in multi-threaded code, then that similarly applies to use of non-reentrant functions but not to other "potentially dangerous" functions.

Reply:
Done.